File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/trace/unresolved/UnresolvedMetaAttribute.java
package org.kevoree.modeling.api.trace.unresolved;

import org.kevoree.modeling.api.meta.MetaType;
import org.kevoree.modeling.api.strategy.ExtrapolationStrategy;
import org.kevoree.modeling.api.meta.MetaAttribute;
import org.kevoree.modeling.api.meta.MetaClass;

/**
 * Created by duke on 10/16/14.
 */
public class UnresolvedMetaAttribute implements MetaAttribute {

    private String metaName;

    public UnresolvedMetaAttribute(String metaName) {
        this.metaName = metaName;
    }

    @Override
    public boolean key() {
        return false;
    }

    @Override
    public MetaClass origin() {
        return null;
    }

    @Override
    public MetaType metaType() {
        return null;
    }

    @Override
    public ExtrapolationStrategy strategy() {
        return null;
    }

    @Override
    public double precision() {
        return 0;
    }

    @Override
    public void setExtrapolationStrategy(ExtrapolationStrategy extrapolationStrategy) {

    }

    @Override
    public String metaName() {
        return metaName;
    }

    @Override
    public int index() {
        return -1;
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/trace/unresolved/UnresolvedMetaAttribute.ts
///<reference path="../../meta/MetaType.ts"/>
///<reference path="../../strategy/ExtrapolationStrategy.ts"/>
///<reference path="../../meta/MetaAttribute.ts"/>
///<reference path="../../meta/MetaClass.ts"/>

class UnresolvedMetaAttribute implements MetaAttribute {

  private _metaName: string = null;

  constructor(p_metaName: string) {
    this._metaName = p_metaName;
  }

  public key(): boolean {
    return false;
  }

  public origin(): MetaClass {
    return null;
  }

  public metaType(): MetaType {
    return null;
  }

  public strategy(): ExtrapolationStrategy {
    return null;
  }

  public precision(): number {
    return 0;
  }

  public setExtrapolationStrategy(extrapolationStrategy: ExtrapolationStrategy): void {
  }

  public metaName(): string {
    return this._metaName;
  }

  public index(): number {
    return -1;
  }

}

